src/random-seed/random-seed.c

```diff
@@ -20,9 +21,10 @@
 #define POOL_SIZE_MAX (10*1024*1024)
 
 int main(int argc, char *argv[]) {
+        _cleanup_free_ struct rand_pool_info *info = NULL;
         _cleanup_close_ int seed_fd = -1, random_fd = -1;
         bool read_seed_file, write_seed_file;
-        _cleanup_free_ void* buf = NULL;
+        unsigned entropy_count = 0;
```

Comment:
hmm, this is weird, why "unsigned"? the kernel structure actually expects an "int" here (which appears kinda sloppy to me, but anyway...).  And philosophically it appears to me this should be a size_t, as ideally if we pass fully random data in then the entropy_count should equal the buffer size, no?

Reply:
Dunno. I started with a size_t, then I think I had some reason to change it, though I don't recall why. The kernel caps it at, I think, 4096 anyway (or some other similar smallish number), so it doesn't matter too much.

Reply:
still, we generally care about type issues in our codebase. If the kernel APIs are a bit sloppy with types then this doesn't mean we should be sloppy too. Please pick the most appropriate type always.